src/ansiblelint/rules/no_changed_when.md

```diff
@@ -1,12 +1,19 @@
 # no-changed-when
 
-This rule checks that tasks return changes to results or conditions.
-Unless tasks only read information, you should ensure that they return changes in the following ways:
+This rule checks that tasks return changes to results or conditions. Unless
+tasks only read information, you should ensure that they return changes in the
+following ways:
 
 - Register results or conditions and use the `changed_when` clause.
 - Use the `creates` or `removes` argument.
 
-You should use the `when` clause to run tasks only if a check returns a particular result.
+You should always use the `changed_when` clause on tasks that do not naturally
+detect if a change has occurred or not. Some of the most common examples are
+[shell] and [command] modules, which run arbitrary commands.
+
+One very common workaround is to use a boolean value like `changed_when: false`
+but you can also use any expressions, including ones that use the registered
+result of a task, like in our example below.
```

Comment:
My minor suggestion would be to expand on the use of boolean values, something like:

```suggestion
if the task never changes anything or `changed_when: true` if it always
changes something, but you can also use any expressions, including ones that
use the registered result of a task, like in our example below.
```
I added the `true` case for completeness but my point was to explain *when* to use the boolean case - it could be condensed to just explaining "if the task never changes anything, but you can also ...". Unfortunately after this addition it is quite a long single sentence - I am sure a better than my wording can be found or possibly split the sentence before the "but".